src/azure-cli/azure/cli/command_modules/apim/custom.py

```diff
@@ -113,3 +114,94 @@ def apim_apply_network_configuration_updates(client, resource_group_name, name,
         properties['location'] = location
 
     return client.api_management_service.apply_network_configuration_updates(resource_group_name, name, properties)
+
+
+# API Operations
+
+def create_apim_api(client, resource_group_name, service_name, api_id, description=None, subscription_key_parameter_names=None,
+                    api_revision=None, api_version=None, is_current=True, display_name=None, service_url=None, protocols=Protocol.https.value, path=None,
+                    api_type=ApiType.http.value, subscription_required=False, tags=None, no_wait=False):
+    """Creates a new API. """
+    resource = ApiContract(
+        api_id=api_id,
+        description=description,
+        subscription_key_parameter_names=subscription_key_parameter_names,
+        api_revision=api_revision,
+        api_version=api_version,
+        is_current=is_current,
+        display_name=display_name,
+        service_url=service_url,
+        protocols=protocols.split(','),
+        path=path,
+        api_type=api_type,
+        subscription_required=subscription_required,
+        tags=tags
+    )
+
+    cms = client.api
+
+    return sdk_no_wait(no_wait, cms.create_or_update,
+                       resource_group_name=resource_group_name,
+                       service_name=service_name, api_id=api_id, parameters=resource)
+
+
+def get_apim_api(client, resource_group_name, service_name, api_id):
+    """Shows details of an API. """
+
+    return client.api.get(resource_group_name, service_name, api_id)
+
+
+def list_apim_api(client, resource_group_name, service_name):
+    """List all APIs of an API Management instance. """
+
+    return client.api.list_by_service(resource_group_name, service_name)
+
+
+def delete_apim_api(client, resource_group_name, service_name, api_id, delete_revisions=True):
+    """Deletes an existing API. """
+
+    return client.api.delete(resource_group_name, service_name, api_id, if_match='*', delete_revisions=delete_revisions)
+
+
+def update_apim_api(instance, description=None, subscription_key_parameter_names=None,
+                    api_revision=None, api_version=None, is_current=True, display_name=None, service_url=None, protocols=Protocol.https.value, path=None,
+                    api_type=ApiType.http.value, subscription_required=False, tags=None):
```

Comment:
For `update` command, it's better not to provide default value for each parameter. Otherwise, it may surprise user by updating some properties in background.

Reply:
Thanks for the hint, I will remove these.